microovn/ovn/services.go

```diff
@@ -0,0 +1,69 @@
+package ovn
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/canonical/microcluster/state"
+	"github.com/canonical/microovn/microovn/database"
+)
+
+func DisableService(s *state.State, service string) error {
+	err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		exists, err := database.ServiceExists(ctx, tx, s.Name(), service)
+		if err != nil {
+			return err
+		}
+		if exists != true {
+			return errors.New("No such service")
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	err = snapStop(service, true)
+	if err != nil {
+		return fmt.Errorf("Snapctl error, likely due to service not existing:\n %w", err)
+	}
+
+	err = s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
+		err := database.DeleteService(ctx, tx, s.Name(), service)
+		return err
+	})
+
+	return err
+
+}
+
+func EnableService(s *state.State, service string) error {
+	err := s.Database.Transaction(s.Context, func(ctx context.Context, tx *sql.Tx) error {
```

Comment:
There's a function that might give you the same functionality as this transaction
https://github.com/canonical/microovn/blob/a99570ddbcffe6acbd76698a424c78d3cca49773/microovn/node/services.go#L41-L43